### PR TITLE
Initial stab at custom CORS headers.

### PR DESCRIPTION
**Pull request by ericholscher**

### CODE REVIEW

File: readthedocs/core/signals.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+from urlparse import urlparse
+
+
+def handler(sender, request, **kwargs):
+    whitelist_urls = ['/api/v2/footer_html', '/api/v2/search']
+    host = urlparse(request.META['HTTP_ORIGIN']).netloc.split(':')[0]
+    domains = Domain.objects.filter(domain__icontains=host)
+    for url in whitelist_urls:
+        if (request.path_info.startswith(url) and domains.count()):
```
Comment (berkerpeksag) [inline readthedocs/core/signals.py]: Perhaps `domains.count()` can be moved outside of the loop? If `domains.count()` returns 0, we can just return `False`.

File: readthedocs/core/apps.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class CoreAppConfig(AppConfig):
+    name = 'readthedocs.core'
+    verbose_name = 'Core'
+
+    def ready(self):
+        if hasattr(self, 'already_run'):
+            return
+        self.already_run = True
```
Comment (agjohnson) [inline readthedocs/core/apps.py]: I've seen this in a few places. What is the purpose of this check? `ready` should only be called once, afaik. Is there a case where it is called multiple times?

Comment (ericholscher) [inline readthedocs/core/apps.py]: Yea, I was wondering about that myself -- it's in all of our `ready` calls, but seems unnecessary.

Comment (agjohnson) [inline readthedocs/core/apps.py]: Ah, here's more info, there's a note that says testing applications could execute `ready` more than once:
https://docs.djangoproject.com/en/1.8/ref/applications/#django.apps.AppConfig.ready

If we do need this, `if hasattr(...)` should be `getattr(self, 'already_run', False)`

Comment (ericholscher) [inline readthedocs/core/apps.py]: Seems our tests are still passing after removing it, so I'll kill it for now.

**Pull request merged**